src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -229,31 +214,43 @@ private void launchNode(SchedulerDriver driver, Offer offer,
   }
 
   private void launchDataNode(SchedulerDriver driver, Offer offer) {
-    launchNode(driver, offer, DATA_NODE_ID, Arrays.asList(DATA_NODE_ID), NODE_EXECUTOR_ID);
+    launchNode(
+        driver,
+        offer,
+        HDFSConstants.DATA_NODE_ID,
+        Arrays.asList(HDFSConstants.DATA_NODE_ID),
+        HDFSConstants.NODE_EXECUTOR_ID);
   }
 
   private void launchInitialJournalNodes(SchedulerDriver driver, Collection<Offer> offers) {
-    int journalNodes = 0;
-    for (Offer offer : offers) {
-      pendingOffers.remove(offer.getId());
-      launchNode(driver, offer, JOURNAL_NODE_ID, Arrays.asList(JOURNAL_NODE_ID), NODE_EXECUTOR_ID);
-      journalNodes++;
-      if (journalNodes >= conf.getJournalNodeCount()) {
-        return;
+    for (int i = 0; i < conf.getJournalNodeCount(); i++) {
+      if (offers.size() > 0) {
```

Comment:
This is the number of journal nodes we launch aside from the 2 journal nodes launched as part of the NN machines. So this could be configured to be 1, 2, etc.